pkg/controller/plan/controller.go

```diff
@@ -263,6 +254,33 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 	return
 }
 
+func (r *Reconciler) archiveAndUpdate(plan *api.Plan) {
```

Comment:
I feel that hiding two method calls behind one makes the code significantly harder to follow.